src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -53,12 +56,12 @@ private ClassName(String packageName, ClassName enclosingClassName, String simpl
   private ClassName(String packageName, ClassName enclosingClassName, String simpleName,
       List<AnnotationSpec> annotations) {
     super(annotations);
-    this.packageName = packageName;
+    this.packageName = (packageName != null) ? packageName : NO_PACKAGE;
```

Comment:
Is this still required? Seems like we might want to prefer exploding instead
```suggestion
    this.packageName = Objects.requireNonNull(packageName, "packageName == null")
```

Reply:
Technically not required - I just wanted to be defensive for the future (I tracked down all first level call-sites, but it's hard to tell)